extendr-macros/src/dataframe.rs

```diff
@@ -0,0 +1,58 @@
+use proc_macro::TokenStream;
+use quote::quote;
+use syn::{parse_macro_input, Data, DataStruct, DeriveInput};
+
+fn parse_struct(input: &DeriveInput, datastruct: &DataStruct) -> TokenStream {
```

Comment:
I think it's worth considering that the derive macro takes an iterable of structs (ie a set of rows) and turns it into a df, whereas the `Into` trait takes a tuple of vectors (ie, a set of columns). These two methods of constructing a df are therefore quite opposite to each other. I would consider renaming the trait into `IntoDataFrameRow` to make this clear.

Reply:
You can always use the `data_frame!` macro for the other.

Rust programmers tend to think in structures of arrays (SOA) and rarely use AOS.

Reply:
The motivation for this was porting the rust-bio library which is mostly AOS.

Reply:
Well now that the tuple derived is dropped, this is no longer inconsistent at least.

Reply:
Actually, no, I still don't like this. I think if you look at a struct definition with `#[derive(IntoDataFrame)]` you will assume that *a single instance of this struct can become a data frame*, when in reality it means *an iterable of this struct can be turned into a data frame*. I get that there's the convention to name the macro after the trait, but we're not implementing the trait directly on the struct anyway. So I'm kind of still in favour of calling this `#[derive(IntoDataFrameRow)]`.